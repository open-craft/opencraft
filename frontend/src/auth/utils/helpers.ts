import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
