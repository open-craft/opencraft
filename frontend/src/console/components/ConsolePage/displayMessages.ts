const messages = {
  editCourses: {
    defaultMessage: 'Edit courses (Studio)',
    description: 'Label to Studio link.'
  },
  notAllowedForStaff: {
    defaultMessage:
      'This page is not accessible by staff users yet. Use <link>this link</link> instead.',
    description: 'Error message for staff users'
  }
};

export default messages;
