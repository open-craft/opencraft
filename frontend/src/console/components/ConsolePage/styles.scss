@import "~styles/theme";

.console-page {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: stretch;

  .title-container {
    margin: 75px;

    @media only screen and (max-width: 600px)
    {
      margin: 30px;
    }
  }

  h1, h2 {
    text-align: center;
    color: #fff;
  }

  h1 {
    font-family: $playfair-font;
    font-weight: bold;
    font-size: 40px;
  }

  h2 {
    font-family: $chivo-font;
    font-size: 20px;
  }

  .console-page-container {
    background-color: #fff;
    padding: 3rem 1rem;
    margin: 0 -15px;
    flex: 1;

    .console-page-content {
      margin: 0 auto;
      max-width: 1100px;
      justify-content: center;

      .customization-form {
        background-color: #f0f5f7;
        padding: 32px;
        margin-bottom: 16px;

        .loading {
          text-align: center;
          margin-top: 80px;
          margin-bottom: auto;
        }

        h2 {
          font-family: $chivo-font;
          font-size: 24px;
          letter-spacing: 0.72px;
          color: $secondary-1;
          text-align: left;
          margin-bottom: 32px;
        }
      }
    }
  }
}



.header-link {
  color: white;

  &:hover {
    color: white;
  }

  .instance-link {
    margin-left: 15px;
  }
}
