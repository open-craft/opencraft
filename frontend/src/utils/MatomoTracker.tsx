import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
