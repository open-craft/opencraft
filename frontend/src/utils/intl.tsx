import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
