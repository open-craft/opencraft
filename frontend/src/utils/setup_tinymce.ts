import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
