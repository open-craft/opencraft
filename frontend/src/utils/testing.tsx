import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
