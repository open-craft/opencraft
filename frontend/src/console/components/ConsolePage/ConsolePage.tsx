import * as React from 'react';
import { RedeploymentToolbar, CustomizationSideMenu } from 'console/components';
import { Row, Col, Container } from 'react-bootstrap';
import { InstanceSettingsModel } from '../models';

import './styles.scss';

interface Props {
  children: React.ReactNode;
  titleExtra?: React.ReactNode;
  instanceSettings?: InstanceSettingsModel;
}

export const ConsolePage: React.FC<Props> = (props: Props) => {
  const instanceName = () => {
      if (props.instanceSettings) {
        let instanceLink = props.instanceSettings.internalDomainName || ""
        if (props.instanceSettings.externalDomainName) {
          instanceLink = props.instanceSettings.externalDomainName
        }
        return (
          <a className="header-link" href={instanceLink}>
            {props.instanceSettings.instanceName}
            <div className="instance-link" />
          </a>
        );
      } else {
        return (
          <div />
        )
      };
  };

  const instanceStudioLink = () => {
      if (props.instanceSettings) {
        let studioLink = props.instanceSettings.internalStudioDomainName || ""
        if (props.instanceSettings.externalStudioDomainName) {
          studioLink = props.instanceSettings.externalStudioDomainName
        }
        return (
          <a className="header-link" href={studioLink}>
            Edit courses (Studio)
          </a>
        );
      } else {
        return (
          <div />
        )
      };
  };

  return (
    <div className="console-page">
      <div className="title-container">
        <h1>{instanceName}</h1>
        <h2>{instanceStudioLink}</h2>
      </div>

      <RedeploymentToolbar />

      <div className="console-page-container">
        <Row className="console-page-content">
          <Container fluid>
            <Row>
              <Col md="4">
                <CustomizationSideMenu />
              </Col>
              <Col md="8" className="customization-form">
                {props.children}
              </Col>
            </Row>
          </Container>
        </Row>
      </div>
    </div>
  );
};
