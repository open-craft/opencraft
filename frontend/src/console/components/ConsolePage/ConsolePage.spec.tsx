import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ConsolePage } from './ConsolePage';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<ConsolePage />).toJSON();
    expect(tree).toMatchSnapshot();
});
