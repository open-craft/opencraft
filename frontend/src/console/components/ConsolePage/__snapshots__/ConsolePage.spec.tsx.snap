// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="content-page"
>
  <div
    className="title-container"
  >
    <h1>
      <a
        className="header-link"
        href="https://courses.opencraft.hosting"
      >
        Wellington High School
        <div
          className="instance-link"
        />
      </a>
    </h1>
    <h2>
      <a
        className="header-link"
        href="https://studio.courses.opencraft.hosting"
      >
        Edit courses (Studio)
      </a>
    </h2>
  </div>
  <div
    className="d-flex justify-content-center align-middle redeployment-toolbar"
  >
    <div
      className="redeployment-nav"
    >
      <div
        className="status-pill"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <span
          className="dot"
        />
        <span
          className="text"
        >
          Status: Up to date
        </span>
      </div>
      <button
        className="float-right loading btn btn-primary btn-lg"
        disabled={false}
        type="button"
      >
        Deploy (10 updates)
      </button>
    </div>
  </div>
  <div
    className="content-page-container"
  >
    <div
      className="content-page-content row"
    >
      <div
        className="container-fluid"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            Menu
          </div>
          <div
            className="col"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
